docs/architecture.md

```diff
@@ -0,0 +1,43 @@
+# Bootleg Architecture Documentation
+
+Disclaimer: The purpose of the work here is to inspire others to try and implement something like what we've described. Building a full working system of this level of complexity would require a non-trivial amount of work. We believe the information here represents our best-guess at what we might build should we have the time/resources to do so. Our hope is that you will find parts of this project inspiring or thought provoking enough to dig in and experiment on your own. We are open to any questions, comments, feedback but keep in mind that this is a work of fiction.

```

Comment:
Probably worth linking to the medium article in here somewhere with the "work of fiction" as the code is real, it's application isn't